crates/qt-build-utils/src/lib.rs

```diff
@@ -390,18 +402,47 @@ impl QtBuild {
         Err(QtBuildError::QtMissing)
     }
 
-    /// Get the output of running `qmake -query var_name`
+    /// Attempt to get the output of running `qmake -query var_name`.
+    pub fn qmake_query_explicit(
+        qmake_executable: &str,
+        var_name: &str,
+    ) -> Result<String, QtBuildError> {
+        let shell_command = format!("{} -query {}", qmake_executable, var_name);
+        let command_output = if is_windows_host() {
+            Command::new("cmd").args(["/C", &shell_command]).output()
+        } else {
+            Command::new("sh").args(["-c", &shell_command]).output()
+        };
+
+        match command_output {
+            Ok(output) => {
+                let output_string: String;
+                match std::str::from_utf8(&output.stdout) {
+                    Ok(s) => output_string = s.trim().to_string(),
+                    Err(e) => {
+                        return Err(QtBuildError::DecodeQmakeOutputFailed(e));
+                    }
+                }
+                if !output.status.success() {
+                    return Err(QtBuildError::QmakeFailed {
+                        stdout: output_string,
+                    });
+                } else if output.stdout.is_empty() {
+                    return Err(QtBuildError::ParseQmakeOutputFailed {
+                        stdout: "".to_string(),
+                    });
+                }
+                Ok(output_string)
+            }
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => Err(QtBuildError::QtMissing),
+            Err(e) => Err(QtBuildError::RunQmakeFailed(e)),
+        }
+    }
+
+    /// Query the qmake executable which was previously
+    /// verified. Panic on error.
     pub fn qmake_query(&self, var_name: &str) -> String {
```

Comment:
note as part of refactorings going on to the build system, this code has moved to https://github.com/KDAB/cxx-qt/blob/96d323343fc8b59617e8902aa6f61360634d5a62/crates/qt-build-utils/src/installation/qmake.rs#L105

The problem itself appears to be that the path to qmake might not be a binary executable but a script? Wonder if we should have a general Command wrapper to always call things via sh or not :thinking: 

Or whether instead the user should have to provide a binary, doesn't seem to tricky to just write a bash/batch script and execute on that?

Reply:
Where this led to failure was with a .bat script on Windows. I don't think there is a way to make a .bat "more executable".  Probably, executable scripts on Linux already work, although I didn't test that.

Thanks for the heads-up, I will adapt to the refactor that was done. My PR so far solved it locally. Would you prefer such a generic Command wrapper, or do you think the current approach in the PR is OK?

I ran into some build failures on your CI I still need to investigate. Locally the PR works on Linux and Windows, with .sh and .bat qmake wrapper scripts.